src/actions/copy_object.rs

```diff
@@ -0,0 +1,151 @@
+use std::time::Duration;
+
+use time::OffsetDateTime;
+use url::Url;
+
+use super::S3Action;
+use crate::actions::Method;
+use crate::signing::sign;
+use crate::{Bucket, Credentials, Map};
+
+/// Create a copy of an object that is already stored in S3, using a `PUT` request.
+///
+/// Find out more about CopyObject from the [AWS API Reference][api]
+///
+/// [api]: https://docs.aws.amazon.com/AmazonS3/latest/API/API_CopyObject.html
```

Comment:
Some actions also mention a few important pieces of information I found from the AWS docs, for this one I think it would be worth to briefly mention some of the footguns with this methods:

* only objects up to 5 GB can be copied using this method
* even if the server returns a 200 response the copy might have failed